### PR TITLE
dotCMS/core#17720 Navigate to first portlet of section when menu is collapsed

### PR DESCRIPTION
**Pull request by hmoreras**

### CODE REVIEW

File: src/app/view/components/dot-navigation/components/dot-nav-item/dot-nav-item.component.scss [hunk 1]
```diff
@@ -9,7 +9,7 @@
         dot-icon,
         dot-sub-nav,
         .dot-nav__item-label {
-            opacity: 0;
+            display: none;
```
Comment (fmontes) [inline src/app/view/components/dot-navigation/components/dot-nav-item/dot-nav-item.component.scss]: Why? this mess up the opacity animation.

File: src/app/view/components/dot-navigation/dot-navigation.component.spec.ts [hunk 2]
```diff
@@ -216,6 +197,17 @@ describe('DotNavigationComponent', () => {
...
+                const firstItem: DebugElement = de.query(By.css('.dot-nav__list-item'));
+                expect(
+                    firstItem.nativeElement.classList.contains('dot-nav__list-item--active')
+                ).toBe(true);
+                const firstMenuLink: DebugElement = firstItem.query(By.css('.dot-nav-sub__link'));
+                expect(
+                    firstMenuLink.nativeElement.classList.contains('dot-nav-sub__link--active')
+                ).toBe(false);
+            });
+
```
Comment (fmontes) [inline src/app/view/components/dot-navigation/dot-navigation.component.spec.ts]: Don't we need a test to check if the routing is happening?

Comment (hmoreras) [reply src/app/view/components/dot-navigation/dot-navigation.component.spec.ts]: that is tested already on line 185 when is collapsed and 211 when is expanded.

File: src/app/view/components/dot-navigation/dot-navigation.component.spec.ts [hunk 3]
```diff
@@ -182,25 +182,6 @@ describe('DotNavigationComponent', () => {
...
-                ).toBe(true);
-                const firstMenuLink: DebugElement = firstItem.query(By.css('.dot-nav-sub__link'));
-                expect(
-                    firstMenuLink.nativeElement.classList.contains('dot-nav-sub__link--actuve')
-                ).toBe(false);
-            });
-
             it('should navigate to portlet when menu is collapsed', () => {
                 expect(dotNavigationService.goTo).toHaveBeenCalledWith('url/link1');
             });
```
Comment (freddyucv) [inline src/app/view/components/dot-navigation/dot-navigation.component.spec.ts]: we should check that the routing is not happening in this case

Comment (hmoreras) [reply src/app/view/components/dot-navigation/dot-navigation.component.spec.ts]: it needs to happen, is when the menu is expanded that don't.

**Comment by fmontes** [comment]
WhatsApp Image 2019-12-18 at 22 16 42
WhatsApp Image 2019-12-18 at 22 16 41
WhatsApp Image 2019-12-18 at 22 16 40

**Pull request merged**